Fix aes-gcm building failures in sample keyprovider

There is some failures as follows when aes_gcm is building:
```
   Compiling aes-gcm v0.9.4
error[E0432]: unresolved import `aes_gcm::aead::KeyInit`
 --> src/enc_mods/sample_enc/crypto.rs:6:27
  |
6 | use aes_gcm::aead::{Aead, KeyInit};
  |                           ^^^^^^^ no `KeyInit` in the root

error[E0599]: the function or associated item `from_slice` exists for struct `GenericArray<u8, AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>>`, but its trait bounds were not satisfied
   --> src/enc_mods/sample_enc/crypto.rs:16:44
    |
16  |     let encryption_key = Key::<Aes256Gcm>::from_slice(key);
    |                                            ^^^^^^^^^^ function or associated item cannot be called on `GenericArray<u8, AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>>` due to unsatisfied trait bounds
    |
   ::: /root/.cargo/registry/src/github.com-1ecc6299db9ec823/aes-gcm-0.9.4/src/lib.rs:174:1
    |
174 | pub struct AesGcm<Aes, NonceSize>
    | --------------------------------- doesn't satisfy `_: ArrayLength<u8>`
    |
    = note: the following trait bounds were not satisfied:
            `AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>: ArrayLength<u8>`

error[E0277]: the trait bound `AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>: ArrayLength<u8>` is not satisfied
   --> src/enc_mods/sample_enc/crypto.rs:16:26
    |
16  |     let encryption_key = Key::<Aes256Gcm>::from_slice(key);
    |                          ^^^^^^^^^^^^^^^^ the trait `ArrayLength<u8>` is not implemented for `AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>`
    |
    = help: the following other types implement trait `ArrayLength<T>`:
              UInt<N, B0>
              UInt<N, B1>
              UTerm
note: required by a bound in `GenericArray`
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/generic-array-0.14.6/src/lib.rs:178:31
    |
178 | pub struct GenericArray<T, U: ArrayLength<T>> {
    |                               ^^^^^^^^^^^^^^ required by this bound in `GenericArray`

error[E0599]: no function or associated item named `new` found for struct `AesGcm` in the current scope
  --> src/enc_mods/sample_enc/crypto.rs:17:29
   |
17 |     let cipher = Aes256Gcm::new(encryption_key);
   |                             ^^^ function or associated item not found in `AesGcm<Aes256, UInt<UInt<UInt<UInt<UTerm, B1>, B1>, B0>, B0>>`
   |
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
6  | use aes_gcm::NewAead;
   |

warning: unused import: `Aead`
```

--- a/sample_keyprovider/Cargo.toml
+++ b/sample_keyprovider/Cargo.toml
@@ -12,7 +12,7 @@ prost = "0.8"
 futures = "0.3.5"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-aes-gcm = "0.9.2"
+aes-gcm = "0.10.1"
 base64 = "0.13.0"
 clap = "2"
 anyhow = "1.0"

